### PR TITLE
Kickoff thanks workflow after tagging

### PR DESCRIPTION
**Pull request by Mark-Simulacrum**
This commit also refactors the github client support to make it possible to send
requests that don't expect a response (like the workflow dispatch trigger).

This hardcodes rust-lang/thanks as the repository we're kicking the workflow off in; that might want to change later or be moved behind a config flag, but it doesn't seem worth it right now -- it could probably run against some other repository, but doing so doesn't seem all that helpful, and switching it on/off in a fine-grained manner is unlikely to be too useful either. Kicking the workflow off in thanks (presuming you have access) is harmless.

### CODE REVIEW

**Comment by jyn514** [comment]
> This commit also refactors the github client

Does it make sense to use an existing GitHub library rather than maintaining our own (here, and I think also in triagebot)? There's https://github.com/XAMPPRocky/octocrab which is by a rust team member.

**Comment by Mark-Simulacrum** [comment]
We could pull in a library, but given the limited amount of functionality we need and my desire to keep the dependency tree down (mostly to reduce need to bump deps often), I think hand rolling the code here is probably our best option. It's not a significant amount of work (and likely a one time investment), either, so overall seems okay to me. Once we have the release functionality more automated we can make a decision as to whether the weight of the glue code is large enough that we should refactor it away.

I think the calculus is a little different in triagebot, where we already pull in tokio and other dependencies, plus are more oriented towards "random contributions" than here, so the benefits of a library make more sense - octocrab is already used in a few handlers there.

File: src/main.rs [hunk 1]
```diff
@@ -623,8 +623,20 @@ impl Context {
...
+            return Ok(());
+        };
+
         if let Some(repo) = self.config.rustc_tag_repository.clone() {
-            self.tag_repository(signer, &repo, rustc_commit)?;
+            self.tag_repository(signer, &mut github, &repo, rustc_commit)?;
+
+            // Once we've tagged rustc, kick off a thanks workflow run.
+            github
+                .token("rust-lang/thanks")?
```
Comment (pietroalbini) [inline src/main.rs:638]: FYI there shouldn't be a need to fetch a new token, as the token is attached to a GitHub App installation, and there can only be one installation of an app for an organization/user. The endpoint that asks for the installation of a repository simply returns the installation of the user/org it's in, and we could reuse the previous token.

Still, it's probably don't worth it to change it in this PR.

**Review by pietroalbini — APPROVED** [review]

File: src/main.rs (hunk 1 above)
Comment (Mark-Simulacrum) [reply src/main.rs:638]: We could, but we're using this as an opportunity to store the repo name alongside the client. In the future we might also restrict these tokens to just manipulate that repository which gives us some added safety against accidentally pushing to rust-lang/rust master or whatever.

**Pull request merged**